pkg/feature/raw_resources.go

```diff
@@ -26,55 +27,80 @@ import (
-		namespace := object.GetNamespace()
+		if k8serr.IsNotFound(errGet) {
+			if errCreate := cli.Create(ctx, target); client.IgnoreAlreadyExists(errCreate) != nil {
+				return fmt.Errorf("failed to create source %s/%s: %w", namespace, name, errCreate)
+			}
 
-		err := cli.Get(ctx, k8stypes.NamespacedName{Name: name, Namespace: namespace}, object.DeepCopy())
-		if client.IgnoreNotFound(err) != nil {
-			return fmt.Errorf("failed to get object %s/%s: %w", namespace, name, err)
+			return nil
 		}
 
-		if err != nil {
-			// object does not exist and should be created
-			if createErr := cli.Create(ctx, object); client.IgnoreAlreadyExists(createErr) != nil {
-				return fmt.Errorf("failed to create object %s/%s: %w", namespace, name, createErr)
-			}
-		}
-		// object exists, check if it is managed
-		isManaged, isAnnotated := object.GetAnnotations()[annotations.ManagedByODHOperator]
+		isManaged, isAnnotated := target.GetAnnotations()[annotations.ManagedByODHOperator]
 		if isAnnotated && isManaged == "true" {
-			// update the object since we manage it
-			if updateErr := cli.Update(ctx, object); updateErr != nil {
-				return fmt.Errorf("failed to update object %s/%s: %w", namespace, name, updateErr)
+			// reconcile the target since we manage it
+			if errUpdate := patchUsingApplyStrategy(ctx, cli, source, target); errUpdate != nil {
+				return fmt.Errorf("failed to update source %s/%s: %w", namespace, name, errUpdate)
 			}
 		}
-		// object exists and is not manged, skip reconcile allowing users to tweak it
+		// source exists and is not manged, skip reconcile allowing users to tweak it
 	}
+
 	return nil
 }
 
 func patchResources(ctx context.Context, cli client.Client, patches []*unstructured.Unstructured) error {
 	for _, patch := range patches {
-		// Convert the individual resource patch to JSON
-		patchAsJSON, err := patch.MarshalJSON()
-		if err != nil {
-			return fmt.Errorf("error converting yaml to json: %w", err)
+		if errPatch := patchUsingMergeStrategy(ctx, cli, patch); errPatch != nil {
+			return errPatch
 		}
+	}
 
-		if err = cli.Patch(ctx, patch, client.RawPatch(k8stypes.MergePatchType, patchAsJSON)); err != nil {
-			return fmt.Errorf("failed patching resource: %w", err)
-		}
+	return nil
+}
+
+// patchUsingApplyStrategy applies a server-side apply patch to a Kubernetes resource.
+// It treats the provided source as the desired state of the resource and attempts to
+// reconcile the target resource to match this state. The function takes ownership of the
+// fields specified in the target and will ensure they match desired state.
+func patchUsingApplyStrategy(ctx context.Context, cli client.Client, source, target *unstructured.Unstructured) error {
+	data, errJSON := source.MarshalJSON()
+	if errJSON != nil {
+		return fmt.Errorf("error converting yaml to json: %w", errJSON)
+	}
+	return cli.Patch(ctx, target, client.RawPatch(k8stypes.ApplyPatchType, data), client.ForceOwnership, client.FieldOwner("rhods-operator"))
```

Comment:
rhods-operator ?

Reply:
Took it from the existing trusted bundle one. As I don't pass any owner here (like with deploy/kustomize) I thought using the same should be fine. The naming is outdated, I agree, but if we want to change it, let's change in both places in other PR.